appinventor/components/src/com/google/appinventor/components/runtime/TextBoxBase.java

```diff
@@ -499,6 +539,30 @@ public void TextColor(int argb) {
     }
   }
 
+  @SimpleFunction(description = "Repositions the cursor of the %type% before the character at the given"
+      + " 1-indexed position. If the given position is larger than the length of the %type%, the cursor will be moved"
+      + " to the end of the text; and if the given position is smaller or equal to 1, the cursor will be moved to the start.")
+  public void MoveCursorTo(int position) {
+    int len = view.getText().toString().length();
+    if (position > len) {
+      view.setSelection(len);
+    } else if (position <= 1) {
+      view.setSelection(0);
+    } else {
+      view.setSelection(position - 1);
+    }
+  }
+
+  @SimpleFunction(description = "Repositions the cursor to the end of the %type%'s text.")
+  public void MoveCursorToEnd() {
+    MoveCursorTo(view.getText().length());
```

Comment:
In testing this, I found that the code moved the cursor to the character before the end. This should add +1 here to account for the -1 in the MoveCursorTo method to account for the 1-indexing of App Inventor

Reply:
Alternatively, change the conditional on line 547 to be `if (position >= len)`

Reply:
I would prefer the former to the latter.

Reply:
Updated.